### ISSUE TITLE
Deadlock when calling AddTarget() with a relative path like <../.xtra_namespace>

### ISSUE THREAD

**Issue by blevin** [PixarAnimationStudios/USD#76]
Daniel Berry reports a deadlock when trying to add a relative path in usd:

https://groups.google.com/d/msg/usd-interest/ZHVas9tPVoY/obsyxSYJGwAJ

This turns out to be a bug in `Usd_InstanceCache::IsPathMasterOrInMaster()`, which does not handle relative paths correctly at the moment.

The repro is easy:

```
from pxr import Usd, Sdf
s = Usd.Stage.CreateInMemory()
p = s.DefinePrim("/ShotObject01")
a = p.CreateAttribute("xtra_namespace", Sdf.ValueTypeNames.String)
a.Set("ShotObject01")

b = s.DefinePrim(p.GetPath().AppendPath("geo"))
r = b.CreateRelationship("xtra_namespace")
r.AddTarget("../.xtra_namespace")
```

**Comment by dbaz** [comment]
Cheers Brett!

**Comment by jtran56** [comment]
Filed as internal issue #138430.

**Comment by sunyab** [comment]
Commit has been merged into release v0.7.2, closing.
